pkg/config/analysis/analyzers/deployment/services.go

```diff
@@ -60,14 +61,18 @@ func (s *ServiceAssociationAnalyzer) Metadata() analysis.Metadata {
 
 func (s *ServiceAssociationAnalyzer) Analyze(c analysis.Context) {
 	c.ForEach(collections.K8SAppsV1Deployments.Name(), func(r *resource.Instance) bool {
-		if util.DeploymentInMesh(r, c) {
+		if util.DeploymentInMesh(r, c) && !isWaypointDeployment(r) {
```

Comment:
nit: suggesting that the added check should be first to fail fast as it is a simple test